chatsky/context_storages/database.py

```diff
@@ -8,181 +8,156 @@
-        :param value: Context to store.
-        """
-        return asyncio.run(self.set_item_async(key, value))
-
     @abstractmethod
-    async def set_item_async(self, key: Hashable, value: Context):
+    async def update_main_info(
+        self, ctx_id: str, turn_id: int, crt_at: int, upd_at: int, misc: bytes, fw_data: bytes
+    ) -> None:
         """
-        Asynchronous method for storing Context.
-
-        :param key: Hashable key used to store Context instance.
-        :param value: Context to store.
+        Update main information about the context storage.
         """
         raise NotImplementedError
 
-    def __delitem__(self, key: Hashable):
-        """
-        Synchronous method for removing stored Context.
-
-        :param key: Hashable key used to identify Context instance for deletion.
-        """
-        return asyncio.run(self.del_item_async(key))
-
     @abstractmethod
-    async def del_item_async(self, key: Hashable):
+    async def delete_context(self, ctx_id: str) -> None:
         """
-        Asynchronous method for removing stored Context.
-
-        :param key: Hashable key used to identify Context instance for deletion.
+        Delete context from context storage.
         """
         raise NotImplementedError
 
-    def __contains__(self, key: Hashable) -> bool:
-        """
-        Synchronous method for finding whether any Context is stored with given key.
-
-        :param key: Hashable key used to check if Context instance is stored.
-        :return: True if there is Context accessible by given key, False otherwise.
-        """
-        return asyncio.run(self.contains_async(key))
-
     @abstractmethod
-    async def contains_async(self, key: Hashable) -> bool:
+    async def load_field_latest(self, ctx_id: str, field_name: str) -> List[Tuple[int, bytes]]:
         """
-        Asynchronous method for finding whether any Context is stored with given key.
-
-        :param key: Hashable key used to check if Context instance is stored.
-        :return: True if there is Context accessible by given key, False otherwise.
+        Load the latest field data.
         """
         raise NotImplementedError
 
-    def __len__(self) -> int:
-        """
-        Synchronous method for retrieving number of stored Contexts.
-
-        :return: The number of stored Contexts.
-        """
-        return asyncio.run(self.len_async())
-
     @abstractmethod
-    async def len_async(self) -> int:
+    async def load_field_keys(self, ctx_id: str, field_name: str) -> List[int]:
         """
-        Asynchronous method for retrieving number of stored Contexts.
-
-        :return: The number of stored Contexts.
+        Load all field keys.
         """
         raise NotImplementedError
 
-    def get(self, key: Hashable, default: Optional[Context] = None) -> Context:
+    @abstractmethod
+    async def load_field_items(self, ctx_id: str, field_name: str, keys: List[int]) -> List[Tuple[int, bytes]]:
```

Comment:
This should return `List[Tuple[int, Optional[bytes]]]`, right?

Add type alias for field items that should be used in all field method signatures.
(this would make it easier to update the type without needing to update signature of every method like with the recent `Optional` addition)